scripts/typeGeneration/convert.js

```diff
@@ -0,0 +1,135 @@
+import { basename, dirname, join, resolve } from 'https://deno.land/std/path/posix.ts';
+import { parse, stringify } from 'https://deno.land/std/yaml/mod.ts';
+import { camelCase } from 'https://deno.land/x/case/mod.ts';
+import { generate } from 'npm:openapi-typescript-codegen';
+import converter from 'npm:swagger2openapi';
+
+const apiTypes = ['management'];
+
+const getFiles = async folder => {
+  const files = [];
+  for await (const singleDirEntry of Deno.readDir(folder)) {
+    if (singleDirEntry.isDirectory) {
+      files.push(...(await getFiles(join(folder, singleDirEntry.name))));
+    } else if (singleDirEntry.isFile) {
+      const res = resolve(folder, singleDirEntry.name);
+      const filename = res.substring(res.lastIndexOf('/') + 1, res.indexOf('.yml'));
+      const apiType = filename.substring(0, filename.indexOf('_'));
+      let versionSuffix = filename.substring(filename.lastIndexOf('_'));
+      versionSuffix = versionSuffix != '_api' ? versionSuffix : '';
+      const service = basename(dirname(res));
+      if (apiTypes.includes(apiType)) {
+        files.push({ filename, apiType, path: res, service, versionSuffix });
+      }
+    }
+  }
+  return files;
+};
+
+const getFileContents = async file => {
+  const fileContent = await Deno.readTextFile(file.path);
+  const fileData = parse(fileContent);
+  if (!fileData.swagger) {
+    return { ...file, fileData };
+  }
+  const converted = await converter.convertObj(fileData, {});
+  return { ...file, fileData: converted.openapi };
+};
+
+const sanitizeSchemaRefs = (pathSpec, renamedSchemas) => {
+  const text = JSON.stringify(pathSpec);
+  const sanitizedText = renamedSchemas.reduce((accu, { key, sanitizedKey }) => accu.replaceAll(`/schemas/${key}"`, `/schemas/${sanitizedKey}"`), text);
+  return JSON.parse(sanitizedText);
+};
+
+const repairPathSpec = (pathSpec, service, versionSuffix) => {
+  const text = JSON.stringify(pathSpec);
+  const sanitizedText = text.replaceAll(`"summary":`, `"description":`);
+  const content = JSON.parse(sanitizedText);
+  return Object.entries(content).reduce(
+    (accu, [key, value]) => ({
+      ...accu,
+      [key]: {
+        ...value,
+        summary: value.summary ?? value.operationId,
+        operationId: camelCase(`${service}${value.operationId || value.summary}${versionSuffix}`)
+      }
+    }),
+    {}
+  );
+};
+
+const sanitizeSchemas = (schemas, existingSchemas, service, counter = {}) =>
+  Object.entries(schemas).reduce(
+    (schemaAccu, [key, schema]) => {
+      let sanitizedKey = key;
+      if (existingSchemas[key]) {
+        schemaAccu.counter[key] = (schemaAccu.counter[key] ?? 1) + 1;
+        console.log(`duplicate schema, candidate for merge: ${key} (${service}) - total: ${schemaAccu.counter[key]}`);
```

Comment:
this in general sums up my doubts about merging all the files: I am afraid that it will be challenging. on the other hand multiple files will bring collisions in the generated code. I have no solution here.

Reply:
IMO the collisions are prevented due to the API paths - so that _should_ not be problematic... and IMO we could handle the deduplication on a different level (e.g. the gui uses the entire API as a single, merged backend in a way 😅)